src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/CLinker.java

```diff
@@ -149,43 +163,43 @@ static CLinker getInstance() {
     MemorySegment upcallStub(MethodHandle target, FunctionDescriptor function);
 
     /**
-     * The layout for the {@code char} C type
+     * The layout for the {@code char} C type, or {@code null} on unsupported platforms.
```

Comment:
Here there are three main options:
1) the one you pursued
2) one where layout constants become instance fields - that's painful for clients because then static imports no longer work
3) turn static fields into static methods (and throw UOE)

All these options are not great IMHO. I think something like (1) might work - but we should consider some alternatives: for instance, we could just say that if a platform is unsupported, the contents of the layout constants is unspecified (once and for all, in the toplevel doc). Then, in the impl, we could either return null, or simply return e.g. SysV layouts, and call it done.

Reply:
Solved by reverting to failing to initialize and added a note about that in the class javadoc.